{"$message_type":"diagnostic","message":"failed to resolve: could not find `error` in `yoshi_std`","code":{"code":"E0433","explanation":"An undeclared crate, module, or type was used.\n\nErroneous code example:\n\n```compile_fail,E0433\nlet map = HashMap::new();\n// error: failed to resolve: use of undeclared type `HashMap`\n```\n\nPlease verify you didn't misspell the type/module's name or that you didn't\nforget to import it:\n\n```\nuse std::collections::HashMap; // HashMap has been imported.\nlet map: HashMap<u32, u32> = HashMap::new(); // So it can be used!\n```\n\nIf you've expected to use a crate name:\n\n```compile_fail\nuse ferris_wheel::BigO;\n// error: failed to resolve: use of undeclared module or unlinked crate\n```\n\nMake sure the crate has been added as a dependency in `Cargo.toml`.\n\nTo use a module from your current crate, add the `crate::` prefix to the path.\n"},"level":"error","spans":[{"file_name":"yoshi\\tests\\type_aliases_tests.rs","byte_start":6077,"byte_end":6082,"line_start":179,"line_end":179,"column_start":63,"column_end":68,"is_primary":true,"text":[{"text":"    assert_eq!(TypeId::of::<Yoshi>(), TypeId::of::<yoshi_std::error::Yoshi>());","highlight_start":63,"highlight_end":68}],"label":"could not find `error` in `yoshi_std`","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"a trait with a similar name exists","code":null,"level":"help","spans":[{"file_name":"yoshi\\tests\\type_aliases_tests.rs","byte_start":6077,"byte_end":6082,"line_start":179,"line_end":179,"column_start":63,"column_end":68,"is_primary":true,"text":[{"text":"    assert_eq!(TypeId::of::<Yoshi>(), TypeId::of::<yoshi_std::error::Yoshi>());","highlight_start":63,"highlight_end":68}],"label":null,"suggested_replacement":"Error","suggestion_applicability":"MaybeIncorrect","expansion":null}],"children":[],"rendered":null}],"rendered":"\u001b[0m\u001b[1m\u001b[38;5;9merror[E0433]\u001b[0m\u001b[0m\u001b[1m\u001b[38;5;15m: failed to resolve: could not find `error` in `yoshi_std`\u001b[0m\n\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m--> \u001b[0m\u001b[0myoshi\\tests\\type_aliases_tests.rs:179:63\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\n\u001b[0m\u001b[1m\u001b[38;5;14m179\u001b[0m\u001b[0m \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m \u001b[0m\u001b[0m    assert_eq!(TypeId::of::<Yoshi>(), TypeId::of::<yoshi_std::error::Yoshi>());\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                                                               \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9m^^^^^\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                                                               \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9m|\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                                                               \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9mcould not find `error` in `yoshi_std`\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                                                               \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9mhelp: a trait with a similar name exists: `Error`\u001b[0m\n\n"}
{"$message_type":"diagnostic","message":"mismatched types","code":{"code":"E0308","explanation":"Expected type did not match the received type.\n\nErroneous code examples:\n\n```compile_fail,E0308\nfn plus_one(x: i32) -> i32 {\n    x + 1\n}\n\nplus_one(\"Not a number\");\n//       ^^^^^^^^^^^^^^ expected `i32`, found `&str`\n\nif \"Not a bool\" {\n// ^^^^^^^^^^^^ expected `bool`, found `&str`\n}\n\nlet x: f32 = \"Not a float\";\n//     ---   ^^^^^^^^^^^^^ expected `f32`, found `&str`\n//     |\n//     expected due to this\n```\n\nThis error occurs when an expression was used in a place where the compiler\nexpected an expression of a different type. It can occur in several cases, the\nmost common being when calling a function and passing an argument which has a\ndifferent type than the matching type in the function declaration.\n"},"level":"error","spans":[{"file_name":"yoshi\\tests\\type_aliases_tests.rs","byte_start":8821,"byte_end":8835,"line_start":265,"line_end":265,"column_start":69,"column_end":83,"is_primary":true,"text":[{"text":"    let _read_fn: fn(&std::path::Path) -> std::io::Result<String> = read_to_string;","highlight_start":69,"highlight_end":83}],"label":"one type is more general than the other","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"yoshi\\tests\\type_aliases_tests.rs","byte_start":8771,"byte_end":8818,"line_start":265,"line_end":265,"column_start":19,"column_end":66,"is_primary":false,"text":[{"text":"    let _read_fn: fn(&std::path::Path) -> std::io::Result<String> = read_to_string;","highlight_start":19,"highlight_end":66}],"label":"expected due to this","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected fn pointer `for<'a> fn(&'a Path) -> std::result::Result<_, _>`\n      found fn item `fn(_) -> std::result::Result<_, _> {yoshi::read_to_string::<_>}`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"\u001b[0m\u001b[1m\u001b[38;5;9merror[E0308]\u001b[0m\u001b[0m\u001b[1m\u001b[38;5;15m: mismatched types\u001b[0m\n\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m--> \u001b[0m\u001b[0myoshi\\tests\\type_aliases_tests.rs:265:69\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\n\u001b[0m\u001b[1m\u001b[38;5;14m265\u001b[0m\u001b[0m \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m \u001b[0m\u001b[0m    let _read_fn: fn(&std::path::Path) -> std::io::Result<String> = read_to_string;\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m-----------------------------------------------\u001b[0m\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9m^^^^^^^^^^^^^^\u001b[0m\u001b[0m \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9mone type is more general than the other\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14mexpected due to this\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m= \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;15mnote\u001b[0m\u001b[0m: expected fn pointer `\u001b[0m\u001b[0m\u001b[1m\u001b[35mfor<'a> \u001b[0m\u001b[0mfn(\u001b[0m\u001b[0m\u001b[1m\u001b[35m&'a Path\u001b[0m\u001b[0m) -> std::result::Result<_, _>`\u001b[0m\n\u001b[0m                  found fn item `fn(\u001b[0m\u001b[0m\u001b[1m\u001b[35m_\u001b[0m\u001b[0m) -> std::result::Result<_, _>\u001b[0m\u001b[0m\u001b[1m\u001b[35m {yoshi::read_to_string::<_>}\u001b[0m\u001b[0m`\u001b[0m\n\n"}
{"$message_type":"diagnostic","message":"mismatched types","code":{"code":"E0308","explanation":"Expected type did not match the received type.\n\nErroneous code examples:\n\n```compile_fail,E0308\nfn plus_one(x: i32) -> i32 {\n    x + 1\n}\n\nplus_one(\"Not a number\");\n//       ^^^^^^^^^^^^^^ expected `i32`, found `&str`\n\nif \"Not a bool\" {\n// ^^^^^^^^^^^^ expected `bool`, found `&str`\n}\n\nlet x: f32 = \"Not a float\";\n//     ---   ^^^^^^^^^^^^^ expected `f32`, found `&str`\n//     |\n//     expected due to this\n```\n\nThis error occurs when an expression was used in a place where the compiler\nexpected an expression of a different type. It can occur in several cases, the\nmost common being when calling a function and passing an argument which has a\ndifferent type than the matching type in the function declaration.\n"},"level":"error","spans":[{"file_name":"yoshi\\tests\\type_aliases_tests.rs","byte_start":8913,"byte_end":8917,"line_start":266,"line_end":266,"column_start":76,"column_end":80,"is_primary":true,"text":[{"text":"    let _read_bytes_fn: fn(&std::path::Path) -> std::io::Result<Vec<u8>> = read;","highlight_start":76,"highlight_end":80}],"label":"one type is more general than the other","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"yoshi\\tests\\type_aliases_tests.rs","byte_start":8862,"byte_end":8910,"line_start":266,"line_end":266,"column_start":25,"column_end":73,"is_primary":false,"text":[{"text":"    let _read_bytes_fn: fn(&std::path::Path) -> std::io::Result<Vec<u8>> = read;","highlight_start":25,"highlight_end":73}],"label":"expected due to this","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected fn pointer `for<'a> fn(&'a Path) -> std::result::Result<_, _>`\n      found fn item `fn(_) -> std::result::Result<_, _> {yoshi::read::<_>}`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"\u001b[0m\u001b[1m\u001b[38;5;9merror[E0308]\u001b[0m\u001b[0m\u001b[1m\u001b[38;5;15m: mismatched types\u001b[0m\n\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m--> \u001b[0m\u001b[0myoshi\\tests\\type_aliases_tests.rs:266:76\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\n\u001b[0m\u001b[1m\u001b[38;5;14m266\u001b[0m\u001b[0m \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m \u001b[0m\u001b[0m    let _read_bytes_fn: fn(&std::path::Path) -> std::io::Result<Vec<u8>> = read;\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                         \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m------------------------------------------------\u001b[0m\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9m^^^^\u001b[0m\u001b[0m \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9mone type is more general than the other\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                         \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                         \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14mexpected due to this\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m= \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;15mnote\u001b[0m\u001b[0m: expected fn pointer `\u001b[0m\u001b[0m\u001b[1m\u001b[35mfor<'a> \u001b[0m\u001b[0mfn(\u001b[0m\u001b[0m\u001b[1m\u001b[35m&'a Path\u001b[0m\u001b[0m) -> std::result::Result<_, _>`\u001b[0m\n\u001b[0m                  found fn item `fn(\u001b[0m\u001b[0m\u001b[1m\u001b[35m_\u001b[0m\u001b[0m) -> std::result::Result<_, _>\u001b[0m\u001b[0m\u001b[1m\u001b[35m {yoshi::read::<_>}\u001b[0m\u001b[0m`\u001b[0m\n\n"}
{"$message_type":"diagnostic","message":"mismatched types","code":{"code":"E0308","explanation":"Expected type did not match the received type.\n\nErroneous code examples:\n\n```compile_fail,E0308\nfn plus_one(x: i32) -> i32 {\n    x + 1\n}\n\nplus_one(\"Not a number\");\n//       ^^^^^^^^^^^^^^ expected `i32`, found `&str`\n\nif \"Not a bool\" {\n// ^^^^^^^^^^^^ expected `bool`, found `&str`\n}\n\nlet x: f32 = \"Not a float\";\n//     ---   ^^^^^^^^^^^^^ expected `f32`, found `&str`\n//     |\n//     expected due to this\n```\n\nThis error occurs when an expression was used in a place where the compiler\nexpected an expression of a different type. It can occur in several cases, the\nmost common being when calling a function and passing an argument which has a\ndifferent type than the matching type in the function declaration.\n"},"level":"error","spans":[{"file_name":"yoshi\\tests\\type_aliases_tests.rs","byte_start":8992,"byte_end":8997,"line_start":267,"line_end":267,"column_start":73,"column_end":78,"is_primary":true,"text":[{"text":"    let _write_fn: fn(&std::path::Path, &[u8]) -> std::io::Result<()> = write;","highlight_start":73,"highlight_end":78}],"label":"one type is more general than the other","suggested_replacement":null,"suggestion_applicability":null,"expansion":null},{"file_name":"yoshi\\tests\\type_aliases_tests.rs","byte_start":8939,"byte_end":8989,"line_start":267,"line_end":267,"column_start":20,"column_end":70,"is_primary":false,"text":[{"text":"    let _write_fn: fn(&std::path::Path, &[u8]) -> std::io::Result<()> = write;","highlight_start":20,"highlight_end":70}],"label":"expected due to this","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected fn pointer `for<'a, 'b> fn(&'a Path, &'b [u8]) -> std::result::Result<_, _>`\n      found fn item `fn(_, _) -> std::result::Result<_, _> {yoshi::write::<_, _>}`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"\u001b[0m\u001b[1m\u001b[38;5;9merror[E0308]\u001b[0m\u001b[0m\u001b[1m\u001b[38;5;15m: mismatched types\u001b[0m\n\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m--> \u001b[0m\u001b[0myoshi\\tests\\type_aliases_tests.rs:267:73\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\n\u001b[0m\u001b[1m\u001b[38;5;14m267\u001b[0m\u001b[0m \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m \u001b[0m\u001b[0m    let _write_fn: fn(&std::path::Path, &[u8]) -> std::io::Result<()> = write;\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m--------------------------------------------------\u001b[0m\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9m^^^^^\u001b[0m\u001b[0m \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9mone type is more general than the other\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\u001b[0m                    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14mexpected due to this\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m|\u001b[0m\n\u001b[0m    \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;14m= \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;15mnote\u001b[0m\u001b[0m: expected fn pointer `\u001b[0m\u001b[0m\u001b[1m\u001b[35mfor<'a, 'b> \u001b[0m\u001b[0mfn(\u001b[0m\u001b[0m\u001b[1m\u001b[35m&'a Path\u001b[0m\u001b[0m, \u001b[0m\u001b[0m\u001b[1m\u001b[35m&'b [u8]\u001b[0m\u001b[0m) -> std::result::Result<_, _>`\u001b[0m\n\u001b[0m                  found fn item `fn(\u001b[0m\u001b[0m\u001b[1m\u001b[35m_\u001b[0m\u001b[0m, \u001b[0m\u001b[0m\u001b[1m\u001b[35m_\u001b[0m\u001b[0m) -> std::result::Result<_, _>\u001b[0m\u001b[0m\u001b[1m\u001b[35m {yoshi::write::<_, _>}\u001b[0m\u001b[0m`\u001b[0m\n\n"}
{"$message_type":"diagnostic","message":"aborting due to 4 previous errors","code":null,"level":"error","spans":[],"children":[],"rendered":"\u001b[0m\u001b[1m\u001b[38;5;9merror\u001b[0m\u001b[0m\u001b[1m\u001b[38;5;15m: aborting due to 4 previous errors\u001b[0m\n\n"}
{"$message_type":"diagnostic","message":"Some errors have detailed explanations: E0308, E0433.","code":null,"level":"failure-note","spans":[],"children":[],"rendered":"\u001b[0m\u001b[1m\u001b[38;5;15mSome errors have detailed explanations: E0308, E0433.\u001b[0m\n"}
{"$message_type":"diagnostic","message":"For more information about an error, try `rustc --explain E0308`.","code":null,"level":"failure-note","spans":[],"children":[],"rendered":"\u001b[0m\u001b[1m\u001b[38;5;15mFor more information about an error, try `rustc --explain E0308`.\u001b[0m\n"}
